client/src/app.js

```diff
@@ -57,45 +91,28 @@ const warnLogout = _.debounce(() => {
   );
 }, 1000);
 
-// afterware for responses
-networkInterface.useAfter([
-  {
-    applyAfterware({ response }, next) {
-      if (!response.ok) {
-        response
-          .clone()
-          .text()
-          .then((bodyText) => {
-            console.log(`Network Error: ${response.status} (${response.statusText}) - ${bodyText}`);
-            next();
-          });
-      } else {
-        let isUnauthorized = false;
-        response
-          .clone()
-          .json()
-          .then(({ errors }) => {
-            if (errors) {
-              console.log('GraphQL Errors:', errors);
-              if (_.some(errors, e => e.message.startsWith('Unauthorized'))) {
-                isUnauthorized = true;
-              }
-            }
-          })
-          .then(() => {
-            if (isUnauthorized) {
-              warnLogout();
-              store.dispatch(logout());
-            }
-            next();
-          });
-      }
-    },
-  },
-]);
+// Hack to get around known issue with networkError not having network header errors
+// https://github.com/apollographql/apollo-link/issues/300
+const logoutLink = onError(({ graphQLErrors }) => {
+  if (graphQLErrors && graphQLErrors[0].message.includes('Unauthorized')) {
```

Comment:
yeah this is not ideal but it'll work for now. We should do some research into how others pass these kinds of errors down and how they parse the result to look for specific things like this. It's weird to me that GraphQL doesn't provide a way to pass specific error codes or something.

Reply:
Apollo does... its just broken. The preferred fix for this issue was with a typecast but it didnt work for me.